pyccel/parser/syntax/openmp.py

```diff
@@ -31,160 +35,686 @@ def __init__(self, **kwargs):
+    def __init__(self, **kwargs):
+        """
+        """
+        self.name = kwargs.pop('name')
+        self.clauses = kwargs.pop('clauses')
+
+        txt = self.name
+        txt += check_get_clauses(self, _valid_simd_clauses, self.clauses)
+
+        self._expr = OmpAnnotatedComment(txt)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> OmpSimdConstruct: expr")
+
+        return self._expr
+
+class OmpMasterConstruct(BasicStmt):
+    """Class representing the master construct."""
+    def __init__(self, **kwargs):
+        """
+        """
+        self.name = kwargs.pop('name')
+
+        self._expr = OMP_Master_Construct(self.name)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> OmpMasterConstruct: expr")
+
+        return self._expr
+
+class OmpMaskedConstruct(BasicStmt):
+    """Class representing a Masked construct."""
+    def __init__(self, **kwargs):
+        """
+        """
+        self.name     = kwargs.pop('name')
+        self.combined = kwargs.pop('combined', None)
+        self.clauses  = kwargs.pop('clauses')
+
+        _valid_clauses = (OmpFilter,)
+        com = None
+        if isinstance(self.combined, OmpTaskloopSimd):
+            self.com = self.combined.expr
+            if 'simd' in self.combined.expr:
+                _valid_clauses = _valid_clauses + _valid_simd_clauses
+            _valid_clauses = _valid_clauses + _valid_taskloop_clauses
+
+        txt = check_get_clauses(self, _valid_clauses, self.clauses)
+
+        self._expr = OMP_Masked_Construct(txt, com)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> OmpMaskedConstruct: expr")
+
+        return self._expr
+
+class OmpSectionsConstruct(BasicStmt):
+    """Class representing a Sections construct."""
+    def __init__(self, **kwargs):
+        """
+        """
+        self.name = kwargs.pop('name')
+        self.clauses = kwargs.pop('clauses')
+
+        txt = self.name
+        txt += check_get_clauses(self, _valid_sections_clauses, self.clauses)
+
+        self._expr = OMP_Sections_Construct(txt)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> OmpSectionsConstruct: expr")
+
+        return self._expr
+
+class OmpSectionConstruct(BasicStmt):
+    """Class representing a Section construct."""
+    def __init__(self, **kwargs):
+        """
+        """
+        self.name = kwargs.pop('name')
+
+        self._expr = OMP_Section_Construct(self.name)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> OmpSectionConstruct: expr")
+
+        return self._expr
+
+class OmpDistributeConstruct(BasicStmt):
+    """Class representing a distribute construct."""
+    def __init__(self, **kwargs):
+        """
+        """
```

Comment:
`__init__` functions don't need docstrings. The class docstring fulfils this role